truss-build/src/truss/build/image.py

```diff
@@ -0,0 +1,165 @@
+from .commands import *
+from typing import Dict, List, Optional, Union
+from pathlib import Path
+from .utils import flatten_str_args, make_pip_install_args
+import shlex
+
+# TODO(bola): support secrets
+
+
+class Image:
+    def __init__(self) -> None:
+        self._base_image = ""
+        self._commands: list[Command] = []
+        self._base_image: Optional[str] = None
+
+    # TODO(support static constructors for different ways of pulling base iamge)
+    # Image.from_dockerfile, from_aws_ecr, from_gcp_artifact_registry, from_registry
+    # also can add common base images: Image.truss_base()....
+
+    def serialize(self) -> str:
+        dockerfile = "# syntax = docker/dockerfile:1\n"  # Support BuildKit
+        dockerfile += "# Auto-generated by truss.build, do not edit!\n"
```

Comment:
Nit: consider making these string literals module- (or class-) private constants (`_SYNTAX_HEADER: ClassVar[str] = "...")`

I really think the less string literals appear somewhere hidden/inline in the depth of code the better...